src/WorkerChannel.ts

```diff
@@ -34,36 +30,14 @@ export class WorkerChannel {
         }
     }
 
-    /**
-     * this hook data will be passed to (and set by) all hooks in all scopes
-     */
-    appHookData: HookData = {};
-    /**
-     * this hook data is limited to the app-level scope and persisted only for app-level hooks
-     */
-    appLevelOnlyHookData: HookData = {};
-    programmingModel?: ProgrammingModel;
-    preInvocationHooks: HookCallback[] = [];
-    postInvocationHooks: HookCallback[] = [];
-    appStartHooks: HookCallback[] = [];
-    appTerminateHooks: HookCallback[] = [];
-    functions: { [id: string]: RegisteredFunction } = {};
-    workerIndexingLocked = false;
-    isUsingWorkerIndexing = false;
-    currentEntryPoint?: string;
-
-    constructor(workerId: string, eventStream: IEventStream, legacyFunctionLoader: ILegacyFunctionLoader) {
+    constructor(workerId: string, eventStream: IEventStream) {
         this.workerId = workerId;
         this.eventStream = eventStream;
-        this.legacyFunctionLoader = legacyFunctionLoader;
-        this.packageJson = {};
     }
 
-    reset(): void {
-        // Ideally this resets all app-related data
-        // That worked is tracked by https://github.com/Azure/azure-functions-nodejs-worker/issues/670
-        this.workerIndexingLocked = false;
-        this.isUsingWorkerIndexing = false;
+    resetApp(): void {
+        this.app = new AppContext();
+        this.app.programmingModel = this.defaultProgrammingModel;

```

Comment:
What if we make the `AppContext` constructor accept a `ProgrammingModel` argument and this line can be moved there? Makes logical sense to me to have such logic in a constructor.

Reply:
That would work for this case (environment reload), but for the other case (worker init) we don't have the defaultProgrammingModel yet when creating the AppContext